src/ConnectedKubernetes/ConnectedKubernetes.Autorest/custom/New-AzConnectedKubernetes.ps1

```diff
@@ -606,13 +577,77 @@ function New-AzConnectedKubernetes {
+        # Create any empty array of IArcAgentryConfigurations.
+        # shortened name to avoid class with type name.
+        #
+        # **NOTE** The Swagger naming does NOT match the names that will be used
+        #          in the final helm values file.  Instead there needs to be an
+        #          explicit mapping which is done in TWO places:
+        #          1. The ConfigDP is able to map the (unprotected) settings but
+        #             does not have access to the protected settings so...
+        #          2. This Powershell script has to perform the mapping for
+        #             protected settings.
+        #
+        #          This DOES mean that code changes are required both in the
+        #          Config DP annd this Powershell script if a new Kubernetes
+        #          feature is added.
+        # !!PDS: This may be wrong - should this be a hash of hashes?  no, it;s a list of hashes amnd hashes!
+        $arcAgentryConfigs = New-Object System.Collections.Generic.List[Microsoft.Azure.PowerShell.Cmdlets.ConnectedKubernetes.Models.Api20240715Preview.ArcAgentryConfigurations]
+
+        # Do not send protected settings to CCRP
+        $combinedKeys = $ConfigurationSetting.Keys + $RedactedProtectedConfiguration.Keys
+        $combinedKeys = $combinedKeys | Get-Unique
+        foreach ($feature in $combinedKeys) {
+            $ArcAgentryConfiguration = [Microsoft.Azure.PowerShell.Cmdlets.ConnectedKubernetes.Models.Api20240715Preview.ArcAgentryConfigurations]@{
+                Feature = $feature
+                Setting = $ConfigurationSetting[$feature]
+            }
+            $arcAgentryConfigs.Add($ArcAgentryConfiguration)
+        }
+
+        # It is possible to set an empty value for these parameters and then
+        # the code above gets skipped but we still need to remove the empty
+        # values from $PSBoundParameters.
+        if ($PSBoundParameters.ContainsKey('ConfigurationSetting')) {
+            $PSBoundParameters.Remove('ConfigurationSetting')
+        }
+        if ($PSBoundParameters.ContainsKey('ConfigurationProtectedSetting')) {
+            $PSBoundParameters.Remove('ConfigurationProtectedSetting')
+        }
+
+        $PSBoundParameters.Add('ArcAgentryConfiguration', $arcAgentryConfigs)
+
         Write-Verbose "Creating 'Kubernetes - Azure Arc' object in Azure"
         $Response = Az.ConnectedKubernetes.internal\New-AzConnectedKubernetes @PSBoundParameters
 
+        # !!PDS: Using this twice so need a function.
+        # $arcAgentryConfigs = New-Object System.Collections.Generic.List[Microsoft.Azure.PowerShell.Cmdlets.ConnectedKubernetes.Models.Api20240715Preview.ArcAgentryConfigurations]
+        $arcAgentryConfigs = New-Object System.Collections.ArrayList
+        foreach ($feature in $combinedKeys) {
+            $ArcAgentryConfiguration = @{
+                "Feature" = $feature
+                "Setting" = $ConfigurationSetting[$feature]
```

Comment:
These cannot both pass?  The feature has either a setting or a protected setting and I would expect a key error for those not in one or the other.